src/main/java/dobby/command/EventCommand.java

```diff
@@ -0,0 +1,62 @@
+package dobby.command;
+
+import java.time.format.DateTimeParseException;
+
+import dobby.DobbyException;
+import dobby.TaskList;
+import dobby.task.Event;
+
+public class EventCommand implements Command {
+
+    protected static final String USAGE = "event _description_ /at dd/mm/yyyy hhhh(optional)";
+
+    @Override
+    public String parseInput(TaskList tasks, String text) throws DobbyException {
+        String message;
+        assert text.startsWith("event") : "Event command must start with event";
+        try {
+            int commandLen = "event".length();
+            text = text.substring(commandLen + 1).trim();
+
+            if (text.indexOf("/at") < 0 && text.length() >= 1) { // non-empty description and /by missing
+                throw new DobbyException("Incorrect usage of command.\nSchedule details cannot be empty. "
+                        + "Please try again.\n  " + USAGE);
+            } else if (text.indexOf("/at") == 0) { // no description, by given
+                throw new DobbyException("Incorrect usage of command.\nDescription cannot be empty. "
+                        + "Please try again.\n  " + USAGE);
+            }
+
+            String at = text.substring(text.indexOf("/at") + 4).trim();
+            String description = text.substring(0, text.indexOf("/at") - 1).trim();
+            Event event = new Event(description, at);
+
+            if (at.lastIndexOf(' ') > 0) {
```

Comment:
Although this part is not exactly arrow headed, I think you could do sth like by.lastIndexOf(' ') > 0 && by.substring(1 + by.lastIndexOf(' ')).length() > 4 . You might also want to abstract the boolean argument into a variable with meaningul name, so that people directly understand what you are checking here. Anyway, amazing code!